mteb/evaluation/evaluators/RetrievalEvaluator.py

```diff
@@ -385,22 +385,60 @@ def encode_corpus(self, corpus: List[Dict[str, str]], batch_size: int, **kwargs)
     def encode(self, sentences: List[str], **kwargs):
         return self.model.encode(sentences, **kwargs)
 
-    def encode_conversations(self, conversations: List[List[str]], **kwargs):
+    def encode_conversations(
+        self, conversations: List[List[str]], batch_size: int, **kwargs
+    ):
         if callable(getattr(self.model, "encode_conversations", None)):
             return self.model.encode_conversations(conversations, **kwargs)
         # otherwise fallback to default implementation
         # TODO: add a warning here
         queries = self.convert_conv_history_to_query(conversations)
-        return self.encode_queries(queries, **kwargs)
+        return self.encode_queries(queries, batch_size=batch_size, **kwargs)
 
     def convert_conv_history_to_query(self, conversations: List[List[str]]) -> str:
         if callable(getattr(self.model, "convert_conv_history_to_query", None)):
             return self.model.convert_conv_history_to_query(conversations)
         return convert_conv_history_to_query(conversations)
 
 
-def convert_conv_history_to_query(conversations: List[List[str]]) -> str:
-    return ["; ".join(conv) for conv in conversations]
+def convert_conv_history_to_query(conversations: List[List[Union[str, dict]]]) -> str:
```

Comment:
This changes looks good to me.  Can you update the Tapioca dataset as well to handle this / ensure it still runs?

Reply:
@orionw good catch - seems i introduced a subtle bug when modifying this function. it has been fixed and topiocqa should work now.



Here's the code:

```python

from mteb import MTEB
from mteb.evaluation.evaluators import DRESModel


class DummyModel:
    def encode(self, texts, *args, **kwargs):
        return texts


evaluation = MTEB(tasks=["TopiOCQA"])
evaluation.load_tasks_data()
sdd = evaluation.tasks[0]

q = sdd.queries["validation"]["205-11"]

dres = DRESModel(model=DummyModel())

assert isinstance(q, list) is True, "Query is not a list"

encoded = dres.encode_conversations([q], batch_size=1)
print(encoded[0])
```

Which should return:

```
what was one of the provisions of the petition of right (1628); No person should be forced to provide a gift, loan or tax without an Act of Parliament.; what does the term "petition of right" refer to?; It is an English constitutional document setting out specific individual protections against the state.; who formulated it?; Charles I; what kinds of provisions does it have?; No person should be forced to provide a gift, loan or tax without an Act of Parliament. No free individual should be imprisoned or detained unless a cause has been shown. Soldiers or members of the Royal Navy should not be billeted in private houses without the free consent of the owner.; which period marked the age of sail for the navy mentioned above?; During the reign of Henry VIII.; do they have an escort fleet?; Yes, there are six Type 45 destroyers and 13 Type 23 frigates in active service.; what is type 45?; It is also known as the D or "Daring" class. It is a class of six guided missile destroyers built for the United Kingdom's Royal Navy.; did its engine face any trouble?; On occasion there have been near-complete power generation failures, temporarily disabling propulsion, power generation for weapons, navigational systems and other purposes, leaving the ships vulnerable to "total electric failure".; how strong was their air defense?; The Royal Navy describes the destroyers' mission as being "to shield the Fleet from air attack". It is equipped with the Sea Viper air-defence system utilizing the SAMPSON active electronically scanned array multi-function radar and the S1850M long-range radar. PAAMS is able to track over 2,000 targets and simultaneously control and coordinate multiple missiles in the air at once, allowing a large number of tracks to be intercepted and destroyed at any given time.; how does the first mentioned radar function?; The SAMPSON multi function radar can detect all types of targets out to a distance of 400 km, and is capable of tracking hundreds of targets at any one time.; who developed it?
```